### PR TITLE
Fix the regression of search conversion promotion in omnibox

### PR DESCRIPTION
**Pull request by simonhong**
Resolves https://github.com/brave/brave-browser/issues/31735

This PR fixes current search conversion UI regressions.
And we'll apply new promotion desigin as a next issue.

W/o this PR, graphic is not fit with it's host.
스크린샷 2023-07-20 131343

After:
스크린샷 2023-07-20 131047

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Launch with `--enable-features=BraveSearchOmniboxBanner`
2. Set non brave search as default search engine
3. Type anything in omnibox and check banner type promotion is shown at last
4. When selected or hovered, UI should be changed
5. Launch with `--enable-features=BraveSearchOmniboxButton`
6. Type anything in omnibox and check button type promotion is shown at second
7. When selected, it should have indicator on left
8. Check banner graphic is fit to banner's height

`OmniboxPromotionTest.*`

~~NOTE: This PR doesn't fix the third item(UI regression) in the issue as we have another UI for search promotion.
Layout regression will be handled together when applying new design.~~ => _**Fixed**_

### CODE REVIEW

**Review by goodov — APPROVED** [review]
chromium_src lgtm

File: chromium_src/components/omnibox/browser/autocomplete_controller.cc [hunk 1]
```diff
@@ -89,9 +89,9 @@ void MaybeAddCommanderProvider(AutocompleteController::Providers& providers,
       !provider_client_->IsOffTheRecord())                            \
     providers_.push_back(new PromotionProvider(provider_client_.get()));
 
-// This sort should be done in the middle of
-// AutocompleteController::UpdateResult() because result should be updated
-// before notifying at the last of UpdateResult().
+// This sort should be done right after AutocompleteResult::SortAndCull() in
+// the AutocompleteController::SortCullAndAnnotateResult() to make our sorting
+// run at last but before notifying.
```
Comment (fallaciousreasoning) [inline chromium_src/components/omnibox/browser/autocomplete_controller.cc:94]: Mega nit:
```suggestion
// run last but before notifying.
```

File: components/omnibox/browser/promotion_unittest.cc [hunk 2]
```diff
@@ -114,26 +156,70 @@ TEST_F(OmniboxPromotionTest, ProfileTest) {
   EXPECT_FALSE(HasPromotionMatch());
 }
 
-TEST_F(OmniboxPromotionTest, PromotionProviderTest) {
-  base::test::ScopedFeatureList feature_list;
-  feature_list.InitAndEnableFeature(
+TEST_F(OmniboxPromotionTest, PromotionEntrySortTest) {
+  const int total_match_count = 6;
```
Comment (fallaciousreasoning) [inline components/omnibox/browser/promotion_unittest.cc:160]: Optional: 
```suggestion
  constexpr int total_match_count = 6;
```

**Review by fallaciousreasoning — APPROVED** [review]
LGTM @simonhong! Sorry, I forgot to submit the review :flushed:

File: chromium_src/components/omnibox/browser/autocomplete_controller.cc (hunk 1 above)
Comment (simonhong) [reply chromium_src/components/omnibox/browser/autocomplete_controller.cc]: Updated.

File: components/omnibox/browser/promotion_unittest.cc (hunk 2 above)
Comment (simonhong) [reply components/omnibox/browser/promotion_unittest.cc]: Updated.

**Review by simonhong — COMMENTED** [review]
@fallaciousreasoning NP! thanks for review always :)

**Pull request merged**